### PR TITLE
Add `Window.mouse_rect`

### PR DESCRIPTION
**Pull request by yunline**
```py
import pygame
from pygame._sdl2 import video

screen = pygame.display.set_mode((640, 480))

win = video.Window.from_display_module()

MOUSE_RECT = (100, 100, 300, 200)
mouse_rect_color = "green"

mouse_restricted = False


def toggle_restricted():
    global mouse_restricted, mouse_rect_color
    mouse_restricted = not mouse_restricted
    if mouse_restricted:
        mouse_rect_color = "red"
        win.mouse_rect = MOUSE_RECT
    else:
        mouse_rect_color = "green"
        win.mouse_rect = None
    print(f"mouse rect: {win.mouse_rect}")


print("Press space key to enable/disable mouse rect")

clock = pygame.time.Clock()
running = True
while running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.KEYDOWN and event.key == pygame.K_SPACE:
            toggle_restricted()

    screen.fill("black")
    pygame.draw.rect(screen, mouse_rect_color, MOUSE_RECT, width=1)
    pygame.display.flip()
    clock.tick(30)
```

### CODE REVIEW

File: docs/reST/ref/sdl2_video.rst [hunk 1]
```diff
@@ -231,6 +231,21 @@
...
+   
+   .. attribute:: mouse_rect
+
+      | :sl:`Get or set the mouse confinement rectangle of the window`
+      | :sg:`mouse_rect -> Rect|None`
+
+      Setting this attribute to a :class:`pygame.Rect` to confine the
+      cursor to the specified area of this window.
+
+      This attribute can be None, representing there is no mouse rect.
```
Comment (MyreMylar) [inline docs/reST/ref/sdl2_video.rst:243]:
```suggestion
      This attribute can be None, meaning that there is no mouse rect.
```

Comment (MyreMylar) [reply docs/reST/ref/sdl2_video.rst:243]: Just a small suggestion to make this sentence have clearer english.

**Review by MyreMylar — APPROVED** [review]
Nice little feature! All looks good to me - bar a little bit of english grammar in the docs.

👍

File: docs/reST/ref/sdl2_video.rst [hunk 2]
```diff
@@ -231,6 +231,21 @@
...
       | :sl:`Get the unique window ID (**read-only**)`
       | :sg:`id -> int`
+   
+   .. attribute:: mouse_rect
+
+      | :sl:`Get or set the mouse confinement rectangle of the window`
+      | :sg:`mouse_rect -> Rect|None`
+
+      Setting this attribute to a :class:`pygame.Rect` to confine the
+      cursor to the specified area of this window.
```
Comment (Matiiss) [inline docs/reST/ref/sdl2_video.rst:241]: Maybe change to: "Setting this attribute to a :class:`pygame.Rect` confines the cursor to the specified area of this window." (just changing a verb)
Also this supports any rect-like object, should that be mentioned?

**Review by Matiiss — APPROVED** [review]
Minor language suggestion, otherwise looks great, love the feature! 👍

On a bit of a side note, should  the tests include a test for warnings in versions where this is not available yet?

File: docs/reST/ref/sdl2_video.rst (hunk 2 above)
Comment (yunline) [reply docs/reST/ref/sdl2_video.rst:241]:
```suggestion
      Setting this attribute to a rect-like object confines the
      cursor to the specified area of this window.
```

**Pull request merged**